src/rules/curlyRule.ts

```diff
@@ -72,6 +73,17 @@ export class Rule extends Lint.Rules.AbstractRule {
         type: "functionality",
         typescriptOnly: false,
         hasFix: true,
+        codeExamples: {
+            pass: dedent`
```

Comment:
IMO this also needs to support multiple examples for different configurations / options.

```
codeExamples: {
  default: {
    pass: "...",
    fail: "...",
  },
  "ignore-same-line": { ... }, 
} 
```

Or maybe just an array with an optional description per object.

Reply:
From my limited experience with the doc portion of the codebase, I think an array would be simplest. So
```ts
codeExamples?: {
  config: string; /* Or... ? */
  pass: string;
  fail: string;
}[]
```
I think an explicit config example would be a good idea from the doc user's perspective, but will probably cause some metadata bloat in some rules. 
```js
{
  "rules": { "no-console": true }
}
```